packages/mdc-ripple/_mixins.scss

```diff
@@ -201,75 +210,96 @@
   }
 }
 
-@mixin mdc-states-hover-opacity($opacity, $query: mdc-feature-all()) {
+@mixin mdc-states-hover-opacity(
+  $opacity, $query: mdc-feature-all(), $ripple-target: "&") {
   $feat-color: mdc-feature-create-target($query, color);
 
   // Background wash styles, for both CSS-only and upgraded stateful surfaces
-  &:hover::before {
-    // Opacity falls under color because the chosen opacity is color-dependent in typical usage
-    @include mdc-feature-targets($feat-color) {
-      opacity: $opacity;
+  &:hover {
+    #{$ripple-target}::before {
+      // Opacity falls under color because the chosen opacity is color-dependent in typical usage
+      @include mdc-feature-targets($feat-color) {
+        opacity: $opacity;
+      }
     }
   }
 }
 
-@mixin mdc-states-focus-opacity($opacity, $has-nested-focusable-element: false, $query: mdc-feature-all()) {
-  $feat-animation: mdc-feature-create-target($query, animation);
-  // Opacity falls under color because the chosen opacity is color-dependent in typical usage
-  $feat-color: mdc-feature-create-target($query, color);
+@mixin mdc-states-focus-opacity(
+  $opacity,
+  $has-nested-focusable-element: false,
+  $query: mdc-feature-all(),
+  $ripple-target: "&") {
 
   // Focus overrides hover by reusing the ::before pseudo-element.
   // :focus-within generally works on non-MS browsers and matches when a *child* of the element has focus.
   // It is useful for cases where a component has a focusable element within the root node, e.g. text field,
   // but undesirable in general in case of nested stateful components.
   // We use a modifier class for JS-enabled surfaces to support all use cases in all browsers.
-  $cssOnlyFocusSelector: if(
-    $has-nested-focusable-element,
-    "&:not(.mdc-ripple-upgraded):focus::before, &:not(.mdc-ripple-upgraded):focus-within::before",
-    "&:not(.mdc-ripple-upgraded):focus::before"
-  );
+  @if $has-nested-focusable-element {
```

Comment:
This looks lot better! 🎉 

Thanks for making this more readable.

Reply:
Thanks for the great suggestion!